src/platforms/esp32/components/avm_builtins/uart_driver.c

```diff
@@ -368,37 +372,68 @@ static void uart_driver_do_write(Context *ctx, term msg)
     }
 
     term result_tuple = term_alloc_tuple(2, ctx);
-    term_put_tuple_element(result_tuple, 0, ref);
+    term_put_tuple_element(result_tuple, 0, term_from_ref_ticks(ref_ticks, ctx));
     term_put_tuple_element(result_tuple, 1, OK_ATOM);
 
     send_message(pid, result_tuple, glb);
 }
 
+static void uart_driver_do_close(Context *ctx, term msg)
+{
+    GlobalContext *glb = ctx->global;
+    struct UARTData *uart_data = ctx->platform_data;
+
+    term pid = term_get_tuple_element(msg, 0);
+    term ref = term_get_tuple_element(msg, 1);
+    uint64_t ref_ticks = term_to_ref_ticks(ref);
+
+    list_remove(&uart_data->listener.listeners_list_head);
+
+    term result_tuple = term_alloc_tuple(2, ctx);
+    term_put_tuple_element(result_tuple, 0, term_from_ref_ticks(ref_ticks, ctx));
```

Comment:
pay attention to GC also here.

Reply:
Added REF_SIZE to call to `memory_ensure_free`